bundles/org.eclipse.osgi.tests/src/org/eclipse/osgi/tests/services/datalocation/BasicLocationTests.java

```diff
@@ -86,10 +96,17 @@ private void checkLocation(Location location, boolean leading, boolean trailing,
 		if (!url.getProtocol().equals("file"))
 			return;
 		assertTrue(url.toExternalForm() + " should " + (trailing ? "" : "not") + " have a trailing slash", url.getFile().endsWith("/") == trailing);
-		if (windows)
+		if (OS.isWindows())
 			assertTrue(url.toExternalForm() + " should " + (leading ? "" : "not") + " have a leading slash", url.getFile().startsWith("/") == leading);
 	}
 
+	private void fail(String message, Throwable exception) {
+		AssertionFailedError error = new AssertionFailedError(message);
+		error.initCause(exception);
+		throw error;
+	}
```

Comment:
fyi: this is a reduced version of the method that was inherited from `CoreTest` before. I kept it instead of replacing all the `fail` calls as the exceptions in those cases are part of the functionality-under-test, so they should make the test fail rather than error on occurrence (see https://github.com/eclipse-equinox/equinox/pull/453#discussion_r1434420881).